src/plugin.ts

```diff
@@ -44,23 +47,32 @@ const plugin: FastifyPluginAsync<RecycleBinOptions> = async (fastify, options) =
     items: { taskManager: itemTaskManager, dbService: itemService },
     itemMemberships: { taskManager: itemMembershipTaskManager },
     taskRunner: runner,
+    db,
   } = fastify;
   const {
-    maxItemsInRequest = 10,
-    maxItemsWithResponse = 5,
+    maxItemsInRequest = MAX_TARGETS_FOR_READ_REQUEST,
+    maxItemsWithResponse = MAX_TARGETS_FOR_MODIFY_REQUEST,
     recycleItemPostHook: postHook,
   } = options;
 
-  const recycledItemTaskManager = new RecycledItemTaskManager();
+  const recycledItemService = new RecycledItemService();
+  const recycledItemTaskManager = new RecycledItemTaskManager(recycledItemService);
   fastify.addSchema(common);
 
-  const removeRecycledItems = async (items, actor, log) => {
-    const filteredItems = await Promise.all(
-      items.map(async (item) => {
-        const isDeleted = await isRecycledItem(item.path, actor, log);
-        return !isDeleted ? item : null;
-      }),
+  const removeRecycledItems = async (items) => {
+    if (!items || !items.length) {
+      return;
+    }
+    // get recycled ids from given item paths and filter them out
+    const recycledItems = await recycledItemService.areDeleted(
+      items.map(({ path }) => path),
+      db.pool,
     );
+    const filteredItems = items.map((item) => {
+      return recycledItems.find(({ path }) => item.path.includes(path)) ? null : item;
+    });
```

Comment:
I'm not sure I understand why the items are filtered here. Isn't `areDeleted` supposed to return the given `items` which have been recycled given a subtree path? I.e. is the `recycledItems` list always equal to `filteredItems`?

Reply:
I'll add details, but mainly `areDeleted` returns the entry in the `recycle_items` which aren't the actual items. Plus, the goal of the function is to filter out these items because it is used in post-hooks (ie: filtering out recycled items from get-children).